src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -134,13 +134,11 @@ private Process LaunchAgentProcess(TestPackage package, Guid agentId)
                     string.Format("The {0} framework is not available", targetRuntime),
                     "framework");
 
-            string agentExePath = GetTestAgentExePath(useX86Agent);
+            string agentExePath = GetTestAgentExePath(targetRuntime, useX86Agent);
```

Comment:
I think this code needs moving down a few lines - it looks like target runtime isn't finalised until the code on line 150. Now we need to know exactly which framework version we want to use before selecting the agent, we need to do that a little later, I think.

Reply:
I didn't notice that. I suspect that line 150 may no longer be needed but I'll have to check. I think we want all the decisions about target runtime to have been made before we get to this point if possible!

Reply:
Hmm... I don't call that method at all in the TestCentric engine. In NUnit it's only called here. Assuming the comment on the method definition is correct, it will only change `targetRuntime` if no version has been specified. We don't have any tests to verify that's what it does, however, so I'll add those first.

The only way the situation can arise is if the user specifies something like `--framework:net` on the command-line. IMO, this should get resolved long before we are at the point of launching a process, but I need to test whether that's true.

IOW this is a __really__ good catch!

Reply:
@ChrisMaddock OK, I did some experimenting. It turns out that RuntimeFrameworkService never passes a runtime type without version (like `net`) to the `TestAgency`. It's all resolved in that service beforehand so `TestAgency` can never change the target at line 150 and the call does nothing.

I checked it out in two ways: Running tests with an argument of `--framework:Net` and stepping through the debug build. That's what I expected, because I knew I had removed this call in the TestCentric engine

Based on that, I'll remove the line. If you're worried, I can add some Guard statements that affirm what we believe: that targetFramework is set, includes a version and is available at the start of the method.

Reply:
Thanks for holding on for this one. I checked, and you're right, I was remembering the code incorrectly. But I'm much happier for double checking!